editoast/src/views/rolling_stocks.rs

```diff
@@ -95,6 +112,142 @@ async fn create(
+            Some(rolling_stock) => rolling_stock,
+            None => return Err(RollingStockError::NotFound { rolling_stock_id }.into()),
+        };
+
+    let FormattedImages {
+        compound_image_height,
+        compound_image_width,
+        images,
+    } = format_images(form.images)?;
+
+    // create compound image
+    let compound_image = create_compound_image(
+        db_pool.clone(),
+        compound_image_height,
+        compound_image_width,
+        images.clone(),
+    )
+    .await?;
+
+    // create livery
+    let rolling_stock_livery = RollingStockLiveryModel {
+        name: Some(form.name.into_inner()),
+        rolling_stock_id: Some(rolling_stock_id),
+        compound_image_id: Some(compound_image.id),
+        ..Default::default()
+    };
+    let rolling_stock_livery: RollingStockLivery =
+        rolling_stock_livery.create(db_pool.clone()).await?.into();
+
+    // create separated images
+    for (index, image) in images.into_iter().enumerate() {
+        let mut w = Cursor::new(Vec::new());
+        image.write_to(&mut w, ImageOutputFormat::Png).unwrap();
+
+        let image = Document::new(String::from("image/png"), w.into_inner())
+            .create(db_pool.clone())
+            .await?;
+        let _ = RollingStockSeparatedImageModel {
+            image_id: Some(image.id.unwrap()),
+            livery_id: Some(rolling_stock_livery.id),
+            order: Some(index.try_into().unwrap()),
+            ..Default::default()
+        }
+        .create(db_pool.clone())
+        .await?;
+    }
+
+    Ok(Json(rolling_stock_livery))
+}
+
+struct FormattedImages {
+    compound_image_height: u32,
+    compound_image_width: u32,
+    images: Vec<DynamicImage>,
+}
+
+fn format_images(mut tmp_images: Vec<TempFile>) -> Result<FormattedImages> {
+    let mut separated_images = vec![];
+    let mut max_height: u32 = 0;
+    let mut total_width: u32 = 0;
+
+    tmp_images.sort_by_key(|f| f.file_name.clone().unwrap());
+
+    for f in tmp_images {
+        let file = f.file.into_file();
+        let mut reader = BufReader::new(file);
+        let mut buffer = vec![];
+        reader.read_to_end(&mut buffer).unwrap();
+
+        let image = ImageReader::new(Cursor::new(buffer))
+            .with_guessed_format()
+            .unwrap();
+
+        let image = match image.decode() {
+            Ok(image) => image,
+            Err(_) => return Err(RollingStockError::CannotReadImage.into()),
+        };
+        max_height = cmp::max(max_height, image.height());
+        total_width += image.width();
+
+        separated_images.push(image);
+    }
+
+    Ok(FormattedImages {
+        compound_image_height: max_height,
+        compound_image_width: total_width,
+        images: separated_images,
+    })
+}
+
+async fn create_compound_image(
+    db_pool: Data<DbPool>,
+    compound_image_height: u32,
+    compound_image_width: u32,
+    images: Vec<DynamicImage>,
```

Comment:
This function could take a `FormattedImages` instead.

Reply:
Done